### PR TITLE
adds findRecord to cloudflare only

### PR DESCRIPTION
**Pull request by MrgSub**
This PR adds `findRecord` function to Cloudflare, currently they're the only provider that supports the feature.

**Use case:** 
Re-running ion would error if a record already exists, it helps if we can check to avoid this error.

### CODE REVIEW

File: pkg/platform/src/components/vercel/dns.ts [hunk 1]
```diff
@@ -89,6 +90,12 @@ export function dns(args: DnsArgs) {
     return caaRecord;
   }
 
+  function findRecord() {
+    throw new Error(
+      `Currentr version of the Vercel adapter does not support finding records.`
```
Comment (ImLunaHey) [inline pkg/platform/src/components/vercel/dns.ts]: typo on `Currentr` -> `Current`

**Comment by jayair** [comment]
Oh interesting. Let me ask the team to take a look.

**Comment by fwang** [comment]
Thanks @MrgSub!

Pulumi/Terraform provides a way to ignore existing records. You can set it through `transform`:
```ts
sst.cloudflare.dns({
  zone: "415e6f4653b6d95b775d350f32119abb",
  transform: {
    record: {
      allowOverwrite: true,
    }
  }
})
```
Does that work for you?

**Comment by fwang** [comment]
we are moving `sst/ion -> sst/sst` and unfortunately this is going to be a messy process because of github's limitations

i'm going to close this PR for now and it can be re-opened in the new `sst/sst` repo that will be ready in the next few days if still relevant

**Pull request closed without merging**